src/components/flowchart/index.js

```diff
@@ -283,26 +281,39 @@ export class FlowChart extends Component {
    */
   updateViewExtents(transform) {
     const { k: scale } = transform || getViewTransform(this.view);
+
     const {
       sidebarWidth = 0,
       metaSidebarWidth = 0,
       codeSidebarWidth = 0,
+      width: chartWidth = 0,
+      height: chartHeight = 0,
     } = this.props.chartSize;
-    const { width = 0, height = 0 } = this.props.graphSize;
+
+    const {
+      width: graphWidth = 0,
+      height: graphHeight = 0,
+    } = this.props.graphSize;
 
     const leftSidebarOffset = sidebarWidth / scale;
     const rightSidebarOffset = (metaSidebarWidth + codeSidebarWidth) / scale;
     const margin = this.MARGIN;
 
+    // Find the relative minimum scale to fit whole graph
+    const minScale = Math.min(
+      chartWidth / (graphWidth || 1),
+      chartHeight / (graphHeight || 1)
+    );
+
     setViewExtents(this.view, {
       translate: {
         minX: -leftSidebarOffset - margin,
-        maxX: width + margin + rightSidebarOffset,
+        maxX: graphWidth + margin + rightSidebarOffset,
```

Comment:
We need to add a test to cover the value of `maxX`

Reply:
This should be covered, could you explain further?